src/components/dialog/dialog-injector.ts

```diff
@@ -0,0 +1,16 @@
+import {Injector} from '@angular/core';
+import {MdDialogRef} from './dialog-ref';
+
+
+/** Custom injector type specifically for instantiating components with a dialog. */
+export class DialogInjector implements Injector {
```

Comment:
Not a blocker, but having an `Injector` with a `set(token, notFoundValue)` method would be a nice addition to angular/core.